stdlib/public/core/Filter.swift

```diff
@@ -203,8 +205,28 @@ public struct LazyFilterCollection<
 
   // TODO: swift-3-indexing-model - add docs
   @warn_unused_result
-  public func next(index: Index) -> Index {
-    return LazyFilterIndex(base: _nextFiltered(index.base))
+  public func next(i: Index) -> Index {
+    return LazyFilterIndex(base: _nextFiltered(i.base))
+  }
+
+  // TODO: swift-3-indexing-model - add docs
+  @warn_unused_result
+  public func advance(i: Index, by n: IndexDistance) -> Index {
+    var index = i.base
+    for _ in 0..<n where index != _base.endIndex {
```

Comment:
Actually, I don't think you need to compare against endIndex.  Collections don't provide any particular guarantee of what will happen in that case.  (Except that it is memory safe, unless the collection is itself unsafe.)